web-js/info-window-button/infoWindow_button.html

```diff
@@ -0,0 +1,171 @@
+        "dojo/domReady!"
+      ], function(
+		on,
+		InfoTemplate,
+		registry,
+		number,
+		Color,
+		SimpleLineSymbol,
+		SimpleMarkerSymbol,
+		GraphicsLayer,
+		TextSymbol,
+		webMercatorUtils,
+		Graphic,
+		Locator,
+        parser,
+        ready,
+        BorderContainer,
+        ContentPane,
+        dom,
+        Map,
+        urlUtils,
+        arcgisUtils,
+        Legend,
+        Scalebar
+      ) {
+
+        parser.parse();
+
+		 var map, location ;
+		 var address ;
+		 var graphicsLayer, graphic_org ;
+		
+		  map = new Map("map", { 
+          basemap: "streets", 
+          center: [-95.273, 38.95],
+          zoom: 14
+        });
+		
+		
+		var locator = new Locator("http://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer");
+
+        var infoTemplate = new InfoTemplate();
+          infoTemplate.setTitle("Location");
+          infoTemplate.setContent("<b>Address :</b>${SingleKey}<br/> <button id='InfoButton' onclick='myFunction();javascript:window.close();'>Label</button>"); 
+		  
+		  
+        var symbol = new SimpleMarkerSymbol(
+          SimpleMarkerSymbol.STYLE_CIRCLE, 
+          15, 
+          new SimpleLineSymbol(
+            SimpleLineSymbol.STYLE_SOLID, 
+            new Color([0, 0, 255, 0.5]), 
+            8
+          ), 
+          new Color([0, 0, 255])
+        );
+
+		  window.myFunction = function(){
+			
+			 var point = location ; 
+			 var symbol_stay = new SimpleMarkerSymbol().setStyle("diamond");
+		      
+				
+			 var graphic = new Graphic(point, symbol_stay);
+			 graphic.attr = address.Address;
+			 
+			 var textSymbol = new TextSymbol() ;
+			 textSymbol.Text =  graphic.attr ; //graphic.Attributes["Address"] ; 
+			 
+			 var textGraphic = new Graphic()  ;
+			 textGraphic.geometry = point ;
+			 textGraphic.symbol = textSymbol ;
+			 
+			 map.graphics.add(graphic);
+			 map.graphics.add(textGraphic) ;
+			 graphicsLayer.remove(graphic_org) ;
+			 map.infoWindow.hide();
+
+		  }
+	
+          var location ;
+		  var graphicsLayer, graphic_org ;
+	 	  
+		 		  
+		  var graphicsLayer = new GraphicsLayer() ;
+		  map.addLayer(graphicsLayer) ;
+		  
+		 // var originAddress = " ";
+		 // var DestinationAddress = "";
+		  
+		  
+		  var locator = new Locator("http://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer");
+		  var infoTemplate = new InfoTemplate();
+          infoTemplate.setTitle("Location");
+          infoTemplate.setContent("<b>Address :</b>${Address}<br/><br/><button id='LabelButton' onclick='myFunction();'>Label it</button>"); 
+		  
+		  locator.on("location-to-address-complete", function (evt) {
+              if (evt.address.address) {
+                  locator.outSpatialReference = map.spatialReference;  //// *** THIS
+                  address = evt.address.address;
+                  location = evt.address.location;  //// *** THIS
+          				 
+				// graphic_org = new Graphic(location, symbol, address, infoTemplate);
```

Comment:
it'd be good to get rid of all the non-informative comments and add a few additional comments to explain the code flow a bit.